### PR TITLE
fix: better errors on failed uploads

### PR DESCRIPTION
**Pull request by markflorkowski**
Addresses #258

### CODE REVIEW

**Comment by markflorkowski** [comment]
@juliusmarminge This is something that I hacked together in ~2 min, we should probably figure out how best to do something like this though.

**Review by juliusmarminge — COMMENTED** [review]
This won't work for the same error in server uploads though

**Comment by markflorkowski** [comment]
> This won't work for the same error in server uploads though

because of DOMParser? We can swap it for some other parser, or I can write a regex to extract the strings we want...

**Comment by juliusmarminge** [comment]
> > This won't work for the same error in server uploads though
> 
> 
> 
> because of DOMParser? We can swap it for some other parser, or I can write a regex to extract the strings we want... 

Yes, it's a browser thing.

Regex could work I guess :)

**Comment by markflorkowski** [comment]
@juliusmarminge here's a regex version for you

**Comment by Mr0Bread** [comment]
Hi @markflorkowski 

Maybe I'm missing something, but it looks like to me that you are extracting the error code despite the fact that it's not used anywhere. 

On the image below error constructor receives parsed message only
image

**Comment by markflorkowski** [comment]
@Mr0Bread Yes, I was playing around with what info is returned from s3 to see what would be most useful. Probably should merge the two and return a formatted string, or pass both into the error (one as cause, one as message). This was not meant to be merged as is, more just getting an idea down so we can figure out the correct way to pass this information to the users.

**Comment by juliusmarminge** [comment]
I added some stuff @markflorkowski

File: packages/uploadthing/src/sdk/utils.ts [hunk 1]
```diff
@@ -83,6 +85,7 @@ export const uploadFilesInternal = async (
         formData.append(key, value);
       });
       formData.append("file", file);
+      formData.set("Policy", "");
```
Comment (markflorkowski) [inline packages/uploadthing/src/sdk/utils.ts:88]: ? @juliusmarminge

Comment (juliusmarminge) [reply packages/uploadthing/src/sdk/utils.ts:88]: Ohhh shoot that was my way of testing a failed upload hahhh. Sorry about thst. Revert

File: packages/uploadthing/src/sdk/utils.ts [hunk 2]
```diff
@@ -94,6 +96,15 @@ export const uploadFilesInternal = async (
       });
 
       if (!s3res.ok) {
+        const text = await s3res.text();
+        const parsed = maybeParseResponseXML(text);
+        console.log({ text, parsed });
```
Comment (juliusmarminge) [inline packages/uploadthing/src/sdk/utils.ts:101]: Rm

```suggestion
```

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**